eclipse-collections-testutils/src/test/java/org/eclipse/collections/impl/test/ExceptionThrownTest.java

```diff
@@ -44,12 +42,12 @@ public void callableException()
     {
         try
         {
-            Verify.assertThrows(NullPointerException.class, new EmptyCallable());
+            Verify.assertThrows(NullPointerException.class, (Callable<?>) EmptyCallable::new);
```

Comment:
Is this a Callable that returns Callable?

Reply:
` private static final class EmptyCallable implements Callable<Void> ` - callable that returns nothing. Funny enough - I can switch to new EmptyCallable() and it detects the type correctly, but not via the ::new expression.

Reply:
The expressions don't have the same type, so it makes sense. Roughly:

- `new EmptyCallable()` is an `EmptyCallable`, or `Callable<Void>`
- `EmptyCallable::new` is a `Callable<EmptyCallable>`, or `Callable<Callable<Void>>`

Reply:
So.. we can override assertThrows in Verify to preserve the signature?

Reply:
I think this change can be reverted. If there's a compiler error, we can probably fix it by adding generics here `Verify.<here>assertThrows(`

Reply:
Indeed - it's been reverted.